plugin/completion.py

```diff
@@ -24,25 +24,25 @@ def run(self, edit: Any, item: dict) -> None:
 
             range = Range.from_lsp(text_edit['range'])
             edit_region = range_to_region(range, self.view)
+            # subtract the typed prefix form the end of the text edit
+            edit_region = sublime.Region(edit_region.begin(), edit_region.end() - len(CompletionHandler.last_prefix))
 
             if insert_text_format == InsertTextFormat.Snippet:
+                self.view.erase(edit, edit_region)
                 self.view.run_command("insert_snippet", {"contents": new_text})
             else:
-                # use current point as the end of the edit region
-                current_point = self.view.sel()[0].begin()
-                edit_range = sublime.Region(edit_region.begin(), current_point)
-                self.view.replace(edit, edit_range, new_text)
+                self.view.replace(edit, edit_region, new_text)
 
                 # move the cursor to the end of the text edit
                 sel = self.view.sel()
                 sel.clear()
-                sel.add(edit_range.begin() + len(new_text))
+                sel.add(edit_region.begin() + len(new_text))
```

Comment:
Do we *really* want to modify the selection? What if there is more than one cursor? Is it possible to make an implementation without touching the selection?

Reply:
Not anymore, I removed that :)